Mathlib/Algebra/MonoidAlgebra/Ideal.lean

```diff
@@ -28,31 +28,34 @@ theorem MonoidAlgebra.mem_ideal_span_of_image [Monoid G] [Semiring k] {s : Set G
     x ∈ Ideal.span (MonoidAlgebra.of k G '' s) ↔ ∀ m ∈ x.support, ∃ m' ∈ s, ∃ d, m = d * m' := by
   let RHS : Ideal (MonoidAlgebra k G) :=
     { carrier := { p | ∀ m : G, m ∈ p.support → ∃ m' ∈ s, ∃ d, m = d * m' }
-      add_mem' := fun x y hx hy m hm => by
+      add_mem' := @fun x y hx hy m hm => by
         classical exact (Finset.mem_union.1 <| Finsupp.support_add hm).elim (hx m) (hy m)
       zero_mem' := fun m hm => by cases hm
-      smul_mem' := fun x y hy m hm =>
-        by
-        replace hm := finset.mem_bUnion.mp (Finsupp.support_sum hm)
-        obtain ⟨xm, hxm, hm⟩ := hm
-        replace hm := finset.mem_bUnion.mp (Finsupp.support_sum hm)
+      smul_mem' := fun x y hy m hm => by
+        classical
+        replace hm := Finset.mem_bunionᵢ.mp (Finsupp.support_sum hm)
+        obtain ⟨xm, -, hm⟩ := hm
+        replace hm := Finset.mem_bunionᵢ.mp (Finsupp.support_sum hm)
         obtain ⟨ym, hym, hm⟩ := hm
-        replace hm := finset.mem_singleton.mp (Finsupp.support_single_subset hm)
+        replace hm := Finset.mem_singleton.mp (Finsupp.support_single_subset hm)
         obtain rfl := hm
-        refine' (hy _ hym).imp fun sm => Exists.imp fun hsm => _
+        -- porting note: changed `Exists.imp` to `And.imp_right` due to change in `∃ x ∈ s`
+        -- elaboration
+        refine' (hy _ hym).imp fun sm p => And.imp_right _ p
```

Comment:
Are we expected to change the proofs, or fix the not-matching-lean-3 mathport output?

@digama0, should this be `#align`ed with a subscript `x` if the latter?

Reply:
It seems to me that in practice we just change the proofs and move on.